lib/cron.js

```diff
@@ -209,24 +216,23 @@
 				var diff = date - start,
 					origDate = new Date(date);
 
-				if (!(date.month() in this.month)) {
+				if (!(date.month() in this.month) && Object.keys(this.month).length!=12) {
 					date.add(1, 'M');
 					date.date(1);
 					date.hours(0);
 					date.minutes(0);
 					date.seconds(0);
 					continue;
 				}
-
-				if (!(date.date() in this.dayOfMonth)) {
+				if (!(date.date() in this.dayOfMonth) && Object.keys(this.dayOfMonth).length!=31) {
```

Comment:
Can you add some testing to ensure that this wont break months that have less than 31 days?

Reply:
@ncb000gt  The logic behind this condition Object.keys(this.dayOfMonth).length!=31   is to  whether the dayOfMonth  is set to "*" in the cronString or  not

Reply:
i would be happy to  create tests to cover cases  coved by this fix